### DISCUSSION TITLE
Apollo CSRF Prevention + Image Uploads

### DISCUSSION THREAD

**Discussion by megafinz** [keystonejs/keystone#7900] [Questions] [answered]
Apollo recommends enabling `csrfPrevention` configuration option for the server: https://www.apollographql.com/docs/apollo-server/security/cors/#preventing-cross-site-request-forgery-csrf

The problem is that when this option is enabled Admin UI file uploads stop working. As far as I can understand, the upload client needs to [meet some criteria](https://www.apollographql.com/docs/apollo-server/data/file-uploads/) (e.g. set `Apollo-Require-Preflight` header) in order to be able to make successful upload requests.

Is there a way to configure or work around that?

**Comment by flexdinesh** [comment, marked as answer]
Tracking as a bug here https://github.com/keystonejs/keystone/issues/7925
